lib/ocpp/v201/charge_point.cpp

```diff
@@ -284,7 +295,18 @@ void ChargePoint::on_session_finished(const int32_t evse_id, const int32_t conne
 }
 
 void ChargePoint::on_meter_value(const int32_t evse_id, const MeterValue& meter_value) {
-    this->evses.at(evse_id)->on_meter_value(meter_value);
+    std::lock_guard<std::mutex> lk(this->meter_value_mutex);
```

Comment:
You only need to lock this in case your writing to `this->meter_value`, not the evse because it has its own lock ;)